crates/sui-graphql-rpc/schema.graphql

```diff
@@ -0,0 +1,823 @@
+# Copyright (c) Mysten Labs, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+# Open Questions
+# --------------
+# - Rules for nullabilit/non-nullability
+# - Pattern for multi-gets
+# - Encoding RPC errors
+# - `type` (vs `scalar`) for Move Types
+# - Pattern for upcasting and downcasting
+
+# Check Completeness Against Existing APIs
+# -------------
+# DONE Read
+# DONE - Objects
+# DONE - Transactions
+# DONE - Checkpoints
+# DONE - Events
+# DONE - ProtocolConfig
+# DONE - Chain Identifier
+# DONE Coin
+# DONE Extended
+# DONE Governance
+# DONE Indexer
+# DONE Move Utils
+# TODO Write
+
+# DONE Port designs from Notion
+# DONE - Display
+# DONE - SuiNS
+# DONE Data Consistency API
+# TODO Flesh out Connections
+# TODO - TotalTransactionBlocks for the PTB connection
+
+schema {
+  query: Query
+  subscription: Subscription
+}
+
+type Query {
+  # First four bytes of the network's genesis checkpoint digest
+  # (uniquely defines the network)
+  chainIdentifier: String!
+
+  # First and last checkpoint that the RPC can answer consistent
+  # queries for.
+  consistentRange: ConsistentRange
+
+  owner(address: SuiAddress!): Owner
+  object(address: SuiAddress!, version: BigInt): Object
+  address(address: SuiAddress!): Address
+
+  epoch(epochId: Int): Epoch
+  protocolConfig(protocolVersion: Int): ProtocolConfig
+
+  checkpoint(filter: CheckpointFilter): Checkpoint
+  checkpoints(order: Order!): CheckpointConnection
+
+  transactionBlock(digest: Bytes!): TransactionBlock
+  transactionBlocks: TransactionBlockConnection
+
+  coinMetadata(coinType: String!): CoinMetadata
+
+  queryObjects(query: ObjectQuery): ObjectQueryConnection
+  queryEvents(filter: EventFilter, order: Order!): EventQueryConnection
+  queryTransactionBlocks(
+    filter: TransactionBlockFilter,
+    order: Order!,
+  ): TransactionBlockQueryConnection
+
+  resolveNameServiceAddress(name: String!): Address
```

Comment:
I think we're missing the reverse lookup for name service.

Reply:
I don't have a good understanding of how suins works, but I'd imagine we could have a type here with an address field that contains additional details about a given name.

Reply:
The reverse lookup is on the `Owner` type (`defaultNameServiceName` and `nameServiceNames`).